core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/EntityStatisticsCacheMonitor.scala

```diff
@@ -0,0 +1,112 @@
+package org.broadinstitute.dsde.rawls.monitor
+
+import akka.actor._
+import akka.pattern._
+import cats.effect.{ContextShift, IO}
+import cats.implicits._
+import com.typesafe.scalalogging.LazyLogging
+import org.broadinstitute.dsde.rawls.dataaccess.SlickDataSource
+import org.broadinstitute.dsde.rawls.monitor.EntityStatisticsCacheMonitor.{EntityStatisticsCacheMessage, HandleBacklog, Sweep}
+import slick.dbio.DBIO
+
+import java.sql.Timestamp
+import java.util.UUID
+import scala.concurrent.{ExecutionContext, Future}
+import scala.concurrent.duration._
+
+object EntityStatisticsCacheMonitor {
+  def props(datasource: SlickDataSource, limit: Int)(implicit executionContext: ExecutionContext, cs: ContextShift[IO]): Props = {
+    Props(new EntityStatisticsCacheMonitorActor(datasource, limit))
+  }
+
+  sealed trait EntityStatisticsCacheMessage
+  case object Sweep extends EntityStatisticsCacheMessage
+  case object HandleBacklog extends EntityStatisticsCacheMessage
+}
+
+class EntityStatisticsCacheMonitorActor(val dataSource: SlickDataSource, val limit: Int)(implicit val executionContext: ExecutionContext, val cs: ContextShift[IO]) extends Actor with EntityStatisticsCacheMonitor with LazyLogging {
+
+  override def preStart(): Unit = {
+    super.preStart()
+    self ! Sweep
+  }
+
+  override def receive = {
+    case Sweep => sweep() pipeTo self
+    case HandleBacklog => handleBacklog() pipeTo self
+  }
+
+}
+
+trait EntityStatisticsCacheMonitor extends LazyLogging {
+
+  implicit val executionContext: ExecutionContext
+  implicit val cs: ContextShift[IO]
+  val dataSource: SlickDataSource
+  val limit: Int
+
+  def sweep() = {
+    val checkFuture = for {
+      recordsToUpdate <- dataSource.inTransaction { dataAccess =>
+        dataAccess.workspaceQuery.listOutdatedEntityCaches(limit)
+      }
+
+      updateResults <- recordsToUpdate.toList.traverse { case (workspaceId, lastModified) =>
+        IO.fromFuture(IO(updateStatisticsCache(workspaceId, lastModified))).map { _ =>
+          (workspaceId, lastModified.getTime, System.currentTimeMillis())
+        }
+      }.unsafeToFuture()
```

Comment:
probably need to mess with the akka receive timeout/inactivity timeout. If `updateStatisticsCache` is slow, the `EntityStatisticsCacheMonitorActor` could see no messages for an extended period and shut itself down due to inactivity.